### PR TITLE
[DEV-6078] Upgrade WP plugin to PHP 8.1

### PR DESCRIPTION
**Pull request by andychute**
**What I did:**
- Upgraded to PHP 8
- Updated composer packages due to PHP 8 upgrade
- Fixed various warnings which were thrown when variables aren't properly checked.
- Increased Wordpress tested version support

**Test scenarios:**
- Test plugin on Wordpress 6.3
- Ensure automated tests function correctly.

### CODE REVIEW

**Review by dgarroway98 — APPROVED** [review]
Approving code changes.  Did not test composer changes.

File: circle.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ jobs:
   build:
      working_directory: /var/www
      docker:
-        - image: retreatguru/php71-apache
+        - image: retreatguru/connect-plugin:php81
```
Comment (cloudEdwards) [inline circle.yml:6]: I'm assuming you made this in our dockerhub. It's a clone of a `php81-apache` image

File: circle.yml [hunk 2]
```diff
@@ -32,6 +32,7 @@ jobs:
               cp .env.default .env
               composer install -d programs-remote-tests/
               chown noroot:www-data /var/www/html
+              su noroot -c '/usr/local/bin/wait-db.sh'
```
Comment (cloudEdwards) [inline circle.yml:35]: Where is this `wait-db.sh`? Is it a CircleCI thing?
I couldn't find it anywhere, but I do see it's being used in `dev up --build`

Comment (cloudEdwards) [reply circle.yml:35]: I found the script, is it a docker thing?

`/usr/local/bin/wait-db.sh`
```
#!/bin/sh
until mysql.sh -e "show databases;" 2> /dev/null > /dev/null
do
    echo -n .
    sleep 1
done
echo
```

Comment (cloudEdwards) [reply circle.yml:35]: ok, I found it. I'm happy now
https://github.com/retreatguru/docker/blob/master/programs/bin/wait-db.sh

File: dev [hunk 3]
```diff
@@ -88,4 +92,9 @@ clean() {
     rm $MARKER
 }
 
+# view logs
+logs() {
+    docker compose logs $1 wordpress
+}
```
Comment (cloudEdwards) [inline dev:98]: What is this doing...., oh I get it now.
The `$1` is the options param like `-f` or `--tail`
And then the service is wordpress, naturally.

https://docs.docker.com/engine/reference/commandline/compose_logs/

File: programs-remote-listings/rs-connect.php [hunk 4]
```diff
@@ -162,15 +162,24 @@ public function set_program_meta()
...
-                echo '<meta property="og:image" content="'.$this->program->photo_details->medium->url.'" />'."\n";
-                echo '<meta property="og:image:width" content="'.$this->program->photo_details->medium->width.'" />'."\n";
-                echo '<meta property="og:image:height" content="'.$this->program->photo_details->medium->height.'" />'."\n";
+
+                if (! empty($this->program->photo_details->medium) && is_object($this->program->photo_details->medium)) {
+                    echo '<meta property="og:image" content="'.$this->program->photo_details->medium->url.'" />'."\n";
+                    echo '<meta property="og:image:width" content="'.$this->program->photo_details->medium->width.'" />'."\n";
+                    echo '<meta property="og:image:height" content="'.$this->program->photo_details->medium->height.'" />'."\n";
+                }
+
```
Comment (cloudEdwards) [inline programs-remote-listings/rs-connect.php:182]:
```suggestion

$medium = $this->program->photo_details->medium ?? null;

if (is_object($medium)) {
    echo '<meta property="og:image" content="' . $medium->url . '" />' . "\n";
    echo '<meta property="og:image:width" content="' . $medium->width . '" />' . "\n";
    echo '<meta property="og:image:height" content="' . $medium->height . '" />' . "\n";
}

```

File: programs-remote-tests/composer.json [hunk 5]
```diff
@@ -1,11 +1,12 @@
 {
-    "require": {},
     "require-dev": {
-        "codeception/codeception": "3.1.3",
+        "codeception/codeception": "^4.1",
         "captbaritone/mailcatcher-codeception-module": "1.*",
         "squizlabs/php_codesniffer": "2.7.*",
         "wimg/php-compatibility": "*",
-        "simplyadmire/composer-plugins" : "@dev"
```
Comment (cloudEdwards) [inline programs-remote-tests/composer.json:8]: I guess were not using this anymore

**Review by cloudEdwards — APPROVED** [review]
looks good, minor code style comment.

File: circle.yml (hunk 1 above)
Comment (andychute) [reply circle.yml:6]: yes, there was a lot of cruft and unnecessary software installed in the previous docker image so I made one specifically for testing the connect plugin

File: programs-remote-tests/composer.json (hunk 5 above)
Comment (andychute) [reply programs-remote-tests/composer.json:8]: Doesn't appear so, and it's an abandoned package, so I just dropped it.

**Pull request merged**